flake8 issues on Python3

Some ORNL builders, e.g. `ornl-ubuntu-builder-2`, have been failing MSlice `master` on Python 3 due to `flake8` checks which pass on Python 2. E.g. https://builds.mantidproject.org/job/mslice-pull_requests-ubuntu-16.04-python3/504/console 

The errors are:

```
mslice/presenters/workspace_manager_presenter.py:31:1: C901 'WorkspaceManagerPresenter.notify' is too complex (15)
mslice/scripting/__init__.py:59:1: C901 'get_algorithm_kwargs' is too complex (14)
```

--- a/mslice/presenters/workspace_manager_presenter.py
+++ b/mslice/presenters/workspace_manager_presenter.py
@@ -21,6 +21,21 @@ class WorkspaceManagerPresenter(WorkspaceManagerPresenterInterface):
         self._workspace_manager_view = workspace_view
         self._main_presenter = None
         self._psd = True
+        self._command_map = {
+            Command.SaveSelectedWorkspaceNexus: lambda: self._save_selected_workspace('.nxs'),
+            Command.SaveSelectedWorkspaceAscii: lambda: self._save_selected_workspace('.txt'),
+            Command.SaveSelectedWorkspaceMatlab: lambda: self._save_selected_workspace('.mat'),
+            Command.RemoveSelectedWorkspaces: self._remove_selected_workspaces,
+            Command.RenameWorkspace: self._rename_workspace,
+            Command.CombineWorkspace: self._combine_workspace,
+            Command.SelectionChanged: self._broadcast_selected_workspaces,
+            Command.Add: self._add_workspaces,
+            Command.Subtract: self._subtract_workspace,
+            Command.SaveToADS: self._save_to_ads,
+            Command.ComposeWorkspace:
+                lambda: self._workspace_manager_view._display_error('Please select a Compose action from the dropdown menu'),
+            Command.Scale: self._scale_workspace,
+            Command.Bose: lambda: self._scale_workspace(is_bose=True)}
 
     def register_master(self, main_presenter):
         assert (isinstance(main_presenter, MainPresenterInterface))
@@ -31,32 +46,8 @@ class WorkspaceManagerPresenter(WorkspaceManagerPresenterInterface):
     def notify(self, command):
         self._clear_displayed_error()
         with show_busy(self._workspace_manager_view):
-            if command == Command.SaveSelectedWorkspaceNexus:
-                self._save_selected_workspace('.nxs')
-            elif command == Command.SaveSelectedWorkspaceAscii:
-                self._save_selected_workspace('.txt')
-            elif command == Command.SaveSelectedWorkspaceMatlab:
-                self._save_selected_workspace('.mat')
-            elif command == Command.RemoveSelectedWorkspaces:
-                self._remove_selected_workspaces()
-            elif command == Command.RenameWorkspace:
-                self._rename_workspace()
-            elif command == Command.CombineWorkspace:
-                self._combine_workspace()
-            elif command == Command.SelectionChanged:
-                self._broadcast_selected_workspaces()
-            elif command == Command.Add:
-                self._add_workspaces()
-            elif command == Command.Subtract:
-                self._subtract_workspace()
-            elif command == Command.SaveToADS:
-                self._save_to_ads()
-            elif command == Command.ComposeWorkspace:
-                self._workspace_manager_view._display_error('Please select a Compose action from the dropdown menu')
-            elif command == Command.Scale:
-                self._scale_workspace()
-            elif command == Command.Bose:
-                self._scale_workspace(is_bose=True)
+            if command in self._command_map.keys():
+                self._command_map[command]()
             else:
                 raise ValueError("Workspace Manager Presenter received an unrecognised command: {}".format(str(command)))
 

--- a/mslice/scripting/__init__.py
+++ b/mslice/scripting/__init__.py
@@ -55,20 +55,28 @@ def generate_script_lines(raw_ws, workspace_name):
             lines += ["ws_{} = mc.{}({})\n".format(ws_name, alg_name, kwargs)]
     return lines
 
+def _parse_prop(prop):
+    pval = prop.value()
+    pname = prop.name()
+    hidden = False
+    output_ws = None
+    if isinstance(pval, string_types):
+        pval = pval.replace("__MSL", "").replace("_HIDDEN", "")
+    if prop.name() == "OutputWorkspace":
+        output_ws = pval.replace(".", "_")
+        if "_HIDDEN" in prop.value():
+            hidden = True
+    return pname, pval, output_ws, hidden
 
 def get_algorithm_kwargs(algorithm, existing_ws_refs):
     arguments = []
     output_ws = None
     for prop in algorithm.getProperties():
         if not prop.isDefault():
-            pval = prop.value()
-            if isinstance(pval, string_types):
-                pval = pval.replace("__MSL", "").replace("_HIDDEN", "")
-            if prop.name() == "OutputWorkspace":
-                output_ws = pval.replace(".", "_")
-                if "_HIDDEN" in prop.value():
-                    arguments += ["store=False"]
-                    continue
+            pname, pval, output_ws, hidden = _parse_prop(prop)
+            if hidden:
+                arguments += ["store=False"]
+                continue
             if algorithm.name() == "Load":
                 if prop.name() == "Filename":
                     arguments += ["{}='{}'".format(prop.name(), pval)]

